src/components/Eth2Diagram.js

```diff
@@ -96,85 +199,175 @@ const PrePhase2 = styled.div`
+        <h2>The Beacon Chain</h2>
         <p>
-          Mainnet will still exist as we know it today – secured by{" "}
-          <Link to="/developers/docs/consensus-mechanisms/pow/">
-            proof-of-work and miners
-          </Link>
+          The Beacon Chain will launch as soon as there is enough ETH in{" "}
+          <Link to="https://launchpad.ethereum.org/">the deposit contract</Link>
           .
         </p>
-        <ButtonLink to="/eth2/shard-chains/">More on shard chains</ButtonLink>
-      </Modal>
-      <Modal isOpen={isBCModalOpen} setIsOpen={setBCModalOpen}>
-        <h2>The Beacon Chain</h2>
         <p>
-          The Beacon Chain will launch as soon as there is enough ETH in the
-          deposit contract.
+          The Beacon Chain will become the conductor of Ethereum, coordinating
+          validators and setting the pace for block creation.
         </p>
         <p>At first, it will exist separately from mainnet.</p>
         <ButtonLink to="/eth2/beacon-chain/">
           More on the Beacon Chain
         </ButtonLink>
+      </>
+    )
+  }
+  if (upgrade === SHARDS) {
+    return (
+      <>
+        <h2>Shard chains</h2>
+        <p>
+          Shards will provide lots of extra data to help increase the amount of
+          transactions mainnet can handle. They'll be coordinated by the Beacon
+          Chain.
+        </p>
+        <p>
+          But all transactions will still rely on Mainnet, which will continue
+          to exist as we know it today – secured by{" "}
+          <Link to="/developers/docs/consensus-mechanisms/pow/">
+            proof-of-work and miners
+          </Link>
+          .
+        </p>
+        <ButtonLink to="/eth2/shard-chains/">More on shard chains</ButtonLink>
+      </>
+    )
+  }
+  if (upgrade === DOCKING) {
+    return (
+      <>
+        <h2>The docking</h2>
+        <p>
+          Mainnet will merge with the{" "}
+          <Link to="/developers/docs/consensus-mechanisms/pos/">
+            proof-of-stake
+          </Link>{" "}
+          system, coordinated by the Beacon Chain.
+        </p>
+        <p>
+          This will turn mainnet into a shard within the new system.{" "}
+          <Link to="/developers/docs/consensus-mechanisms/pow/mining/">
+            Miners
+          </Link>{" "}
+          will no longer be needed as all of Ethereum will be secured by{" "}
+          <Link to="/glossary/#validator">validators</Link>.
+        </p>
+        <ButtonLink to="/eth2/docking/">More on the docking</ButtonLink>
+      </>
+    )
+  }
+  return null
+}
+
+// TODO update z-indices so only the selected section of diagram is above the modal overlay
+const Eth2Diagram = () => {
+  const [modalState, setModalState] = useState(false)
+
+  // Update modal state to upgrade props, else close modal
+  const handleClick = (upgrade) => {
+    const newState = upgrade || false
+    setModalState(newState)
+  }
+
+  const themeContext = useContext(ThemeContext)
```

Comment:
@ryancreatescopy I don't believe importing `ThemeContext` is needed here if all we're doing is changing colors based on the dark/light theme.

Instead, we should be creating new variables in the `lightColors` & `darkColors` of theme.js so that depending on dark/light mode, the variable displays the desired colors.

This will remove the needs to pass `isDarkTheme` to all the components & the need for any conditional logic within the styled components.

LMK if this doesn't make sense!